openedx/core/djangoapps/programs/utils.py

```diff
@@ -463,12 +463,22 @@ def _collect_one_click_purchase_eligibility_data(self):
         if is_learner_eligible_for_one_click_purchase:
             for course in self.data['courses']:
                 add_course_sku = False
+                applicable_seat = False
```

Comment:
shouldn't this be in the course_run loop?

Reply:
Yes, good catch.  Making this change and testing now.  Thank you.